snapcraft/lxd.py

```diff
@@ -68,6 +68,7 @@ def _create_container(self):
             yield
         finally:
             check_call(['lxc', 'stop', self._container_name])
+            check_call(['lxc', 'delete', self._container_name])
```

Comment:
It is strange that this is needed as the container is ephemeral so should be destroyed on `stop`